ROADMAP.md

```diff
@@ -0,0 +1,118 @@
+-  E.2 🧪 Future-proof testing - **Comprehensive testing for non-transport features**
+-  G 📢 Judicious Address Advertisements
+
+# Roadmap Appendix
+## [A.⚡️Handshakes at the Speed of Light](https://github.com/libp2p/go-libp2p/issues/1807)
+
+**Why**: Historically, libp2p has been very wasteful when it comes to round trips spent during connection establishment. This is slowing down our users, especially their TTFB (time to first byte) metrics.
+
+**Goal**: go-libp2p optimizes its handshake latency up to the point where only increasing the speed of light would lead to further speedups. In particular, this means:
+
+1. [Early Muxer Negotiation](https://github.com/libp2p/specs/issues/426): cutting off the 1 RTT wasted on muxer negotiation
+2. [Adding security protocol](https://github.com/libp2p/specs/pull/353): cutting off the 1 RTT wasted on security protocol negotiation by including the security protocol in the multiaddr
+3. 0.5 RTT data optimization: using 0.5-RTT data (for TLS) / a Noise Extension to ship the list of Identify protocols, cutting of 1 RTT that many protocols spend waiting on `IdentifyWait`
+
+## [B. 🧠 Smart Dialing](https://github.com/libp2p/go-libp2p/issues/1808)
+
+**Why**: Having a large list of transports to pick from is great. Having an advanced stack that can dial all of them is even greater. But dialing all of them at the same time wastes our, the network’s and the peer’s resources. 
+
+**Goal**: When given a list of multiaddrs of a peer, go-libp2p is smart enough to pick the address that results in the most performant connection (for example, preferring QUIC over TCP), while also picking the address such that maximizes the likelihood of a successful handshake.
+
+How:
+
+1. Happy Eyeballs: implement some kind of “Happy-Eyeballs” style prioritization among all supported transports
+2. QUIC Blackhole detector: detection of blackholes, especially relevant to detect UDP (QUIC) blackholing
+3. RTT estimation: estimation of the expected RTT of a connection based on two nodes’ IP addresses, so that Happy Eyeballs Timeouts can be set dynamically
+
+## [C. 📊 Comprehensive Metrics](https://github.com/libp2p/go-libp2p/issues/1356)
+
+**Why**: For far too long, go-libp2p has been a black box. This has hurt us many times, by allowing trivial bugs to go undetected for a long time ([example](https://github.com/ipfs/kubo/pull/8750)). Having metrics will allow us to track the impact of performance improvements we make over time.
+
+**Goal**: Export a wider set of metrics across go-libp2p components and enable node operators to monitor their nodes in production. Optionally provide a sample Grafana dashboard similar to the resource manager dashboard.
+
+**How**: This will look similar to how we already expose resource manager metrics. Metrics can be added incrementally for libp2p’s components. First milestone is having metrics for the swarm.
+
+## [D. 📺 Universal Browser Connectivity](https://github.com/libp2p/go-libp2p/issues/1811)
```

Comment:
I know the ordering of milestones is defined above. Maybe in a future PR we move the themes around to be in priority order. For example, as a theme, I think browser connectivity should be before all of these. I would likely put future proof testing at the top.